homeassistant/components/mobile_app/http_api.py

```diff
@@ -44,6 +45,20 @@ def __init__(self, store: Store) -> None:
         """Handle the POST request for registration."""
         hass = request.app['hass']
 
+        if ATTR_APP_COMPONENT in data:
+            component = get_component(hass, data[ATTR_APP_COMPONENT])
+            if component is None:
+                fmt_str = "{} is not a valid component."
+                msg = fmt_str.format(data[ATTR_APP_COMPONENT])
+                return self.json_message(msg, HTTP_BAD_REQUEST)
```

Comment:
We should not use `json_message`, it's bad. 

Do something like the websocket response:

```ts
{
  success: False,
  error: {
    code: 'invalid_component',
    message: 'Component does not exist'
  }
}
```

Reply:
and let's standardize on this format for all mobile_app interactions.